### PR TITLE
fix(EMI-2091): spacing and overflow issues and curly quote

### PR DESCRIPTION
**Pull request by rquartararo**
The type of this PR is: **Fix**

This PR solves [EMI-2091]

### Description

This adjusts the spacing of the `Curators’ Pick` and `Increased Interest` labels to give a bit more room underneath the the label while keeping the same 22px line height consistent with the other lines. 

After 
Screenshot 2024-09-17 at 5 41 17 PM

Before
Screenshot 2024-09-17 at 5 41 46 PM

This also fixes the wrapping issue on very narrow width artworks. 

After
Screenshot 2024-09-18 at 10 51 42 AM

Before
Screenshot 2024-09-18 at 10 31 56 AM

This also updates the `Curators’ Pick` label to use a curly quote

Screenshot 2024-09-19 at 12 26 01 PM

[EMI-2091]: https://artsyproduct.atlassian.net/browse/EMI-2091?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCollectorSignal.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const ArtworkSidebarCollectorSignal: React.FC<ArtworkSidebarCollectorSign
         <VerifiedIcon color="black100" width={18} mr={1} mt={0.5} />
         <Stack gap={0}>
           <Text variant="sm" color="black100">
-            Curators' Pick
+            Curators’ Pick
```
Comment (oxaudo) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCollectorSignal.tsx:39]: 🦅

File: src/Apps/Artwork/Components/ArtworkSidebar/__tests__/ArtworkSidebarDetails.jest.tsx [hunk 2]
```diff
@@ -265,7 +265,7 @@ describe("ArtworkSidebarDetails", () => {
         }),
       })
 
-      expect(screen.queryByText(/Curators' Pick/)).toBeInTheDocument()
+      expect(screen.queryByText(/Curators’ Pick/)).toBeInTheDocument()
```
Comment (oxaudo) [inline src/Apps/Artwork/Components/ArtworkSidebar/__tests__/ArtworkSidebarDetails.jest.tsx:268]: Oh. I assume this change should also be done on mobile, right?

**Review by oxaudo — APPROVED** [review]
Looks good!

File: src/Apps/Artwork/Components/ArtworkSidebar/__tests__/ArtworkSidebarDetails.jest.tsx (hunk 2 above)
Comment (MrSltun) [reply src/Apps/Artwork/Components/ArtworkSidebar/__tests__/ArtworkSidebarDetails.jest.tsx:268]: :ceiling-cat: This is already there in Eigen :)

**Review by MrSltun — APPROVED** [review]
Looks good to me! 🌟

**Pull request merged**